### ISSUE TITLE
Error: There was an error turning the model into an object.

### ISSUE THREAD

**Issue by ogrotten** [balderdashy/sails#2986]
sailsjs 0.11, node 0.12.4. 
with passport 0.2.1 and passport-local 1.0
 
Using the sailsjs blueprint routes just to see what's going on. The project is just a simple blog, and I tried looking at a user object in the blueprint route `/user/4` to show me userid 4. The title error message was the result.

Here's a copy of the post I made to SE with errors and a bit of relevant code: https://gist.github.com/ogrotten/6e93ac7197821bdd0e77#file-gistfile1-txt.

After inserting more object outputs, I discover that `_locals: [object Object]` appears when requesting `/user/4`, but not when requesting `/user` which appears to loop over this same function however many times (and outputs a json obj to the webpage).

**Comment by tjwebb** [comment]
I use passport and passport-local in [**sails-auth**](https://github.com/tjwebb/sails-auth) and haven't run into this issue. It sounds like it might be an issue with the way passport is serializing and de-serializing objects. 

Give sails-auth a shot and see if that works.

**Comment by ogrotten** [comment]
In my learning phases, I only discovered sails-(generate-)auth a couple days ago after I was well into this bit here. I'll try it and see what shakes out.

**Comment by DrMoriarty** [comment]
I have the same error on one of my servers. I discovered that the problem is in waterline package. With waterline 0.10.23 there is en error. But with waterline 0.10.18 there aren't any errors.
Check your waterline package version in node_modules/sails/node_modules/waterline/package.json

**Comment by nickautomatic** [comment]
In case it helps anyone, I was having this problem, and it turned out to be fixable by rendering the User object to JSON before trying to render it. Full details are on Stack Overflow: http://stackoverflow.com/a/31025238/180500

**Comment by sailsbot** [comment]
Thanks for posting, @ogrotten.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
